### PR TITLE
feat(api): add custom redirect back

### PR DESCRIPTION
**Pull request by raisedadead**

### CODE REVIEW

File: api-server/server/component-passport.js [hunk 1]
```diff
@@ -112,7 +116,8 @@ export const createPassportCallbackAuthenticator = (strategy, config) => (
       if (!user || !userInfo) {
         return res.redirect('/signin');
       }
-      const redirect = `${homeLocation}/learn`;
+      const redirect = `${returnTo}`;
+      console.log('REDIRECT', redirect);
```
Comment (raisedadead) [inline api-server/server/component-passport.js]: !!!!

Comment (ojeytonwilliams) [reply api-server/server/component-passport.js]: I respect the all caps console logging, though. It's the only way.

Comment (raisedadead) [reply api-server/server/component-passport.js]: You can't miss it.

**Comment by raisedadead** [comment]
One of the most, jankiest ever PR. However, we need this before @QuincyLarson 's makes the big announcement.

Can @freeCodeCamp/dev-team give me a sanity check on this? I am going to push this to beta to test thoroughly.

**Comment by ahmadabdolsaheb** [comment]
tested locally LGTM.

File: api-server/server/utils/middleware.js [hunk 2]
```diff
@@ -56,10 +56,14 @@ export function ifNotVerifiedRedirectToUpdateEmail(req, res, next) {
 }
 
 export function ifUserRedirectTo(path = `${homeLocation}/`, status) {
-  status = status === 302 ? 302 : 301;
+  status = status === 301 ? 301 : 302;
   return (req, res, next) => {
     const { accessToken } = getAccessTokenFromRequest(req);
     if (req.user && accessToken) {
+      if (req.query && req.query.returnTo) {
+        console.log(req.query.returnTo);
```
Comment (ahmadabdolsaheb) [inline api-server/server/utils/middleware.js]: do we need this one?

Comment (ahmadabdolsaheb) [reply api-server/server/utils/middleware.js]: the console.log

**Pull request merged**